src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs

```diff
@@ -573,22 +583,25 @@ LIMIT 1
 
         var cd = new CommandDefinition(
             commandText: @"
-WITH variables (entity_id, ord) AS (
-    SELECT
-        a.val AS entity_id,
-        a.ord AS ord
-     FROM UNNEST(@entityIds) WITH ORDINALITY AS a(val, ord)
+WITH variables (entity_id) AS (
+    SELECT UNNEST(@entityIds)
+),
+metadata_slices AS (
+    SELECT variables.entity_id, emah.metadata_slice, emah.metadata_total_count
+    FROM variables
+    INNER JOIN LATERAL (
```

Comment:
i'm not sure if we're not losing ordering after that join, saw that you're applying that schema with ordinality in next select statement but isn't that too late?

Reply:
I think we're good here. I believe we have a test that validates that?

Reply:
I'm not sure that we have test that validates order of returned metadata, hmmm can you check with Theo?